async-http-client/src/main/scala/org/htt4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -0,0 +1,102 @@
+package org.htt4s.client.asynchttpclient
+
+import com.ning.http.client.AsyncHandler.STATE
+import com.ning.http.client.generators.InputStreamBodyGenerator
+import com.ning.http.client.{Request => AsyncRequest, Response => AsyncResponse, _}
+
+import org.http4s._
+import org.http4s.client._
+import org.http4s.util.task
+import scodec.bits.ByteVector
+
+import scala.collection.JavaConverters._
+
+import scalaz.stream.io._
+import scalaz.stream.async
+import scalaz.concurrent.Task
+
+import scala.concurrent.Promise
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object AsyncHttpClient {
+  val EOF = ByteVector(-1)
```

Comment:
I think a `ByteVector` of this form could come down the wire and therefor this cannot be considered an EOF signal. Maybe an empty `ByteVector` would work? You just need to ensure you don't enqueue and empty one off the wire.